include/nnstreamer-edge.h

```diff
@@ -78,8 +78,9 @@ typedef enum {
   NNS_EDGE_FLAG_RECV = (1 << 0),
   NNS_EDGE_FLAG_SEND = (1 << 1),
   NNS_EDGE_FLAG_SERVER = (1 << 2),
+  NNS_EDGE_FLAG_PUBSUB = (1 << 3),
```

Comment:
What does this do?
(NNS_EDGE_FLAG_PUBSUB? not NNS_EDGE_FLAG_PUB, NNS_EDGE_FLAG_SUB?)

Reply:
It is to distinguish between query and pub/sub.
How about these to make it clear?
```
  NNS_EDGE_FLAG_QUERY_CLIENT = (1 << 0),
  NNS_EDGE_FLAG_QUERY_SERVER = (1 << 1),
  NNS_EDGE_FLAG_PUB = (1 << 2),
  NNS_EDGE_FLAG_SUB = (1 << 3),
```
+@jaeyun-jung

Reply:
> It is to distinguish between query and pub/sub. How about these to make it clear?
> 
> ```
>   NNS_EDGE_FLAG_QUERY_CLIENT = (1 << 0),
>   NNS_EDGE_FLAG_QUERY_SERVER = (1 << 1),
>   NNS_EDGE_FLAG_PUB = (1 << 2),
>   NNS_EDGE_FLAG_SUB = (1 << 3),
> ```
> 
> +@jaeyun-jung

Yes, ```{ QUERY_CLIENT, QUERY_SERVER, PUB, SUB }``` is intuitive and clear.

Reply:
And be aware that ```nns_edge_flag_e``` is NEVER used in this .h file. (a useless interface)
You'll need to draw class (struct/interface) diagrams of this .h file, its users, and its (internal) implementations and review the architecture. If you think the implementation deadline is too imminent for such architecture review & refactoring, you can postpone architecture review & refactoring after the M2 release.

Reply:
`nns_edge_flag_e` is used for creating the edge handle.
```
int nns_edge_create_handle (const char *id, nns_edge_connect_type_e connect_type, int flags, nns_edge_h *edge_h);
```

Reply:
> `nns_edge_flag_e` is used for creating the edge handle.
> 
> ```
> int nns_edge_create_handle (const char *id, nns_edge_connect_type_e connect_type, int flags, nns_edge_h *edge_h);
> ```

Then, change ```int flags``` --> ```nns_edge_flag_e flags```.

Reply:
@anychhoice , Please update the flags and data type.

Reply:
```
typedef enum {
  NNS_EDGE_ROLE_QUERY_CLIENT = 0,
  NNS_EDGE_ROLE_QUERY_SERVER,
  NNS_EDGE_ROLE_PUB,
  NNS_EDGE_ROLE_SUB,

  NNS_EDGE_ROLE_UNKNOWN,
} nns_edge_role_e;
```

How about change name to `nns_edge_role_e` ?

Reply:
Nice suggestion. But I think `nns_edge_node_type_e` is more appropriate.